packages/@aws-cdk/cdk/lib/cloudformation/stack.ts

```diff
@@ -48,6 +49,10 @@ export class Stack extends Construct {
     return (construct as any)._isStack;
   }
 
+  public static artifactIdForStack(stackName: string) {
+    return `${stackName}.template.json`;
```

Comment:
This name should be based off of the construct path, not off of any name the stack gets deployed under.

In the Apps PR, users will be able to control the stack name (might make sense to deploy as the same name in different regions, why not leave it up to them?) but conflicting stack (deployment) names there will overwrite files.

Reply:
Makes sense. I will fix.